parser/src/main/scala/hmda/parser/fi/lar/LarCsvParser.scala

```diff
@@ -83,63 +87,67 @@ object LarCsvParser {
+          agencyCode,
+          loan,
+          preapprovals,
+          actionType,
+          actionDate,
+          geography,
+          applicant,
+          purchaserType,
+          denial,
+          rateSpread,
+          hoepaStatus,
+          lienStatus
+        )
+      )
     } else {
-      Left(larErrors)
+      val lErrors = validation.toEither.left.get
+      Left(lErrors.list.toList)
     }
   }
 
-  def checkLar(s: String): List[String] = {
-    val values = s.split('|').map(_.trim)
-    var errors = List[String]()
-    val numericFields = Map(
-      0 -> "Record Identifier",
-      2 -> "Agency Code",
-      5 -> "Loan Type",
-      6 -> "Property Type",
-      7 -> "Loan Purpose",
-      8 -> "Owner Occupancy",
-      9 -> "Loan Amount",
-      10 -> "Preapprovals",
-      11 -> "Type of Action Taken",
-      12 -> "Date of Action",
-      17 -> "Applicant Ethnicity",
-      18 -> "Co-applicant Ethnicity",
-      19 -> "Applicant Race: 1",
-      24 -> "Co-applicant Race: 1",
-      29 -> "Applicant Sex",
-      30 -> "Co-applicant Sex",
-      32 -> "Type of Purchaser",
-      37 -> "HOEPA Status",
-      38 -> "Lien Status"
-    )
-
-    if (values.length != 40 && values.length != 39) {
-      return errors ::: List("Incorrect number of fields: " + values.length)
+  def toIntorFail(value: String, fieldName: String): ValidationNel[String, List[Int]] = {
+    Try(value.toInt) match {
+      case Failure(result) => s"$fieldName is not an Integer".failure.toValidationNel
+      case Success(result) => List(result).success
     }
+  }
 
-    for (int <- numericFields.keys) {
-      if (Try(values(int).toInt).isFailure) {
-        val field = numericFields(int)
-        errors = errors ::: List(s"$field is not an Integer")
-      }
-    }
+  def checkLar(fields: List[String]): ValidationNel[String, List[Int]] = {
+
+    if (fields.length != 40 && fields.length != 39) {
+      ("Incorrect number of fields: " + fields.length).failure.toValidationNel
+    } else {
+      val numericFields = ListMap(
+        "Record Identifier" -> fields(0),
+        "Agency Code" -> fields(2),
+        "Loan Type" -> fields(5),
+        "Property Type" -> fields(6),
+        "Loan Purpose" -> fields(7),
+        "Owner Occupancy" -> fields(8),
+        "Loan Amount" -> fields(9),
+        "Preapprovals" -> fields(10),
+        "Type of Action Taken" -> fields(11),
+        "Date of Action" -> fields(12),
+        "Applicant Ethnicity" -> fields(17),
+        "Co-applicant Ethnicity" -> fields(18),
+        "Applicant Race: 1" -> fields(19),
+        "Co-applicant Race: 1" -> fields(24),
+        "Applicant Sex" -> fields(29),
+        "Co-applicant Sex" -> fields(30),
+        "Type of Purchaser" -> fields(32),
+        "HOEPA Status" -> fields(37),
+        "Lien Status" -> fields(38)
+      )
 
-    errors
+      val validationList = numericFields.map { case (key, value) => toIntorFail(value, key) }
+
+      validationList.reduce(_ +++ _)
```

Comment:
These two lines are a lot cleaner than the `for` loop, but I'm worried about the overhead and complexity/readability that's added with this implementation.